driver-async/src/main/com/mongodb/async/client/MongoClientImpl.java

```diff
@@ -69,6 +68,7 @@ public MongoDatabase getDatabase(final String name) {
 
     @Override
     public void close() {
+        serverSessionPool.close();
```

Comment:
I don't have a problem with this.  I think that cluster.close() just below also closes network connections synchronously.